c10/core/Scalar.h

```diff
@@ -49,21 +60,32 @@ class C10_API Scalar {
       typename std::enable_if<std::is_same<T, bool>::value, bool>::type* =
           nullptr>
   Scalar(T vv) : tag(Tag::HAS_b) {
-    v.i = convert<int64_t, bool>(vv);
+    v.u.i = convert<int64_t, bool>(vv);
+  }
+
+#define DEFINE_ACCESSOR(type, name)                                     \
+  type to##name() const {                                               \
+    if (Tag::HAS_d == tag) {                                            \
+      return checked_convert<type, double>(v.u.d, #type);               \
+    } else if (Tag::HAS_z == tag) {                                     \
+      return checked_convert<type, c10::complex<double>>(v.u.z, #type); \
+    }                                                                   \
+    if (Tag::HAS_b == tag) {                                            \
+      return checked_convert<type, bool>(v.u.i, #type);                 \
+    } else if (Tag::HAS_i == tag) {                                     \
+      return checked_convert<type, int64_t>(v.u.i, #type);              \
+    } else if (Tag::HAS_si == tag) {                                    \
+      TORCH_CHECK(false)                                                \
```

Comment:
It would be more informative for these to be `TORCH_INTERNAL_ASSERT` as they represent internal invariant violation in our code (and bugs should be filed)